protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+use aptos_sdk::{
+	move_types::language_storage::TypeTag,
+	rest_client::{Client, FaucetClient},
+	types::LocalAccount,
+};
+use aptos_types::account_address::AccountAddress;
+use bridge_shared::{
+	bridge_contracts::{
+		BridgeContractCounterparty, BridgeContractCounterpartyError,
+		BridgeContractCounterpartyResult,
+	},
+	types::{
+		Amount, BridgeTransferDetails, BridgeTransferId, HashLock, HashLockPreImage,
+		InitiatorAddress, RecipientAddress, TimeLock,
+	},
+};
+use rand::prelude::*;
+use serde::Serialize;
+use std::str::FromStr;
+use url::Url;
+
+use crate::utils::MovementAddress;
+
+pub mod utils;
+
+const DUMMY_ADDRESS: AccountAddress = AccountAddress::new([0; 32]);
+const COUNTERPARTY_MODULE_NAME: &str = "atomic_bridge_counterparty";
+
+enum Call {
+	Lock,
+	Complete,
+	Abort,
+	GetDetails,
+}
+
+pub struct MovementClient {
+	///Address of the counterparty moduke
+	counterparty_address: AccountAddress,
+	///Address of the initiator module
+	initiator_address: Vec<u8>,
+	///The Apotos Rest Client
+	rest_client: Client,
+	faucet_client: FaucetClient,
+	///The signer account
+	signer: LocalAccount,
+}
+
+impl MovementClient {
+	pub async fn build_with_config() -> Result<Self, anyhow::Error> {
+		let dot_movement = dot_movement::DotMovement::try_from_env().unwrap();
+		let suzuka_config =
+			dot_movement.try_get_config_from_json::<suzuka_config::Config>().unwrap();
+		let node_connection_address = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_rest_connection_hostname;
+		let node_connection_port =
+			suzuka_config.execution_config.maptos_config.client.maptos_rest_connection_port;
+
+		let node_connection_url =
+			format!("http://{}:{}", node_connection_address, node_connection_port);
+		let node_connection_url = Url::from_str(node_connection_url.as_str()).unwrap();
+
+		let faucet_listen_address = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_faucet_rest_connection_hostname;
+		let faucet_listen_port = suzuka_config
+			.execution_config
+			.maptos_config
+			.client
+			.maptos_faucet_rest_connection_port;
+		let faucet_url = format!("http://{}:{}", faucet_listen_address, faucet_listen_port);
+		let faucet_url = Url::from_str(faucet_url.as_str()).unwrap();
+
+		let rest_client = Client::new(node_connection_url.clone());
+		let faucet_client = FaucetClient::new(faucet_url, node_connection_url.clone());
```

Comment:
What is the necessity of integrating this into our client-side models?

Reply:
We will need a faucet for the tests, so that a user can get tokens to deploy.